### PR TITLE
Harmonize Grids naming

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR harmonizes naming for Grid services.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Please check that nothing is broken on Categories listing, Employees listing and Contacts listing pages

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13189)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks few questions:
1. I'd like to move all query builders from `Grid/Query` to `Grid/Query/Builder` namespace, but that would mean BC break for services introduced in 1.7.5. However, these services are `final` and would be accessible by same `key` in container. Am I allowed to do this "break" here?
2. Kind of related to question above: Am I allowed to change some class naming for some grid services introduced in 1.7.5? Some of the have suffix `*GridDefinitionFactory` while others have `*DefinitionFactory`.
3. Move `Grid/Position` out of `Grid` namespace, this subcomponent has nothing to do with Grid. Should be done in another PR, wdyt?

File: src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ services:
     prestashop.core.grid.data_provider.contacts:
         class: '%prestashop.core.grid.data.factory.doctrine_grid_data_factory%'
         arguments:
-            - '@prestashop.core.grid.query_builder.contacts'
+            - '@prestashop.core.grid.query_builder.contact'
             - '@prestashop.core.hook.dispatcher'
             - '@prestashop.core.grid.query.doctrine_query_parser'
             - 'contacts'
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml]: what we could harmonize is the id of the grids too, but it wont be necessary in 1.7.7 as I'll will suggest a new DataCollector will all the information about the displayed grids.

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
> Move Grid/Position out of Grid namespace, this subcomponent has nothing to do with Grid. 

You can't move but you can deprecate and move => Backward compatible.

I agree that Position component has nothing to do with Grid :)

>  Am I allowed to change some class naming for some grid services introduced in 1.7.5

You can deprecate them and create new ones with a better naming: in the end as we are probably the only ones consumers of these classes, this shouldn't worry anyone.

> I'd like to move all query builders from Grid/Query to Grid/Query/Builder namespace, but that would mean BC break for services introduced in 1.7.5.

Same here, "deprecate and duplicate" is the key if you want to provide backward compatible changes :+1:

**Review by mickaelandrieu — COMMENTED** [review]
I can't approve that fast, let's discuss the scope of this contribution first :-)

**Comment by sarjon** [comment]
> You can't move but you can deprecate and move => Backward compatible.

It was **not** part of 1.7.5, so we move it now or never. :smiling_imp: 

> Same here, "deprecate and duplicate" is the key if you want to provide backward compatible changes

I want to, but I'm wondering if I have to. :stuck_out_tongue:

File: src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml (hunk 1 above)
Comment (sarjon) [reply src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml]: Yes, I'll do that too. :+1:

**Comment by sarjon** [comment]
@matks ready for review. :)

**Comment by matks** [comment]
I'm sorry but I come with the flag of frustration 😞

I dont see a lot of value in doing 1 and 2. Namespace structure and folders structure must respond to a need: need to gather together classes related, need to ease the finding of classes, need to ease the grouping of classes. Namespaces `Grid/Query` to `Grid/Query/Builder` are almost identical to me, and the Builder piece of info is carried by the class names. I dont think this is worth the outcome, creating deprecated old classes for this in a minor version.

In a major version, why not. It's a good moment to reorganize your code. But deprecating/moving code has a cost for both the maintainer and the user, and here the cost seems too high compared to the outcome.

For refactoring 3 "Move Grid/Position out of Grid namespace" it makes more sense, but I think it should be part of a bigger work that is "move Grid component out of Core and make it a stand-alone component" (cc @mickaelandrieu ). Same here, I suggest we list it in PS 8 todo-list and perform the change at this moment where class changes and moves cost nothing and people expect such changes.

Imagine the opposite: tomorrow Symfony decides that Symfony\Component\HttpFoundation is actually not relevant, and Symfony\Component\HttpTools is better. Do you think that deprecating classes from the HttpFoundation and copying them into HttpTools is relevant in a minor version ? This means that people using these classes need to update their code to use the new classes ... for no obvious benefit: no better feature, no optimization, no better security, only better naming according to them 😉

**Comment by sarjon** [comment]
Ok @matks, but regarding `Position` in Grid it will haunt us until next major. :ghost:

**Comment by matks** [comment]
> Ok @matks, but regarding `Position` in Grid it will haunt us until next major. 👻

I'm fine with that level of haunting 😄 currently I'm being haunted by [this](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Addon/Theme/ThemeManager.php#L438) and I find it a lot more terrifying 👻

**Comment by sarjon** [comment]
> I'm fine with that level of haunting smile

Yeah, but again, it's one more _broken window_ (instead of "fixed window") in PrestaShop codebase. :/

Is there anything left for this PR?

**Comment by matks** [comment]
> > I'm fine with that level of haunting smile
> 
> Yeah, but again, it's one more _broken window_ (instead of "fixed window") in PrestaShop codebase. :/
> 

This is not a broken window, the code is OK, the behavior is OK, the naming is relevant it's just not _in the best location_. So I'd say it's a window that could benefit from being relocated somewhere it's more sunny to improve how the house is under sun exposition 😉 .

Right now, it's good. Naming is relevant. Could be better, but it's not bad.

--------

Just need to review now that you removed the WIP 😄

**Comment by sarjon** [comment]
> This is not a broken window

I could argue about it. :smile: Here's why, someone will see this and decide to add another class to Grid, and then another and so on, while we say "ohh, it's fine, it's just a location so let's merge it and go on" until suddenly Grid component becomes huge and unmaintainable with a lot of irrelevant classes. 

That's why I consider it a _broken window_, if we don't fix these small issues early, they will pile up and at some point it will be too late. :)

**Comment by matks** [comment]
> > This is not a broken window
> 
> I could argue about it. 😄 Here's why, someone will see this and decide to add another class to Grid, and then another and so on, while we say "ohh, it's fine, it's just a location so let's merge it and go on"

Not on my watch 😉 .

I's true that that if code does not undergo regular cleaning and refactoring, it rots and ends up into a pile of spaghetti legacy code very hard to maintain. And it's true that keeping the code in a "no issues" (no broken windows) state prevents lazy people to downgrade the code quality. However a lot of these issues do not happen with proper discipline. I believe we can do it.

Even the best codebase in the world can rot without discipline or persistence. Even the worst codebase in the world can improve with discipline and persistence. We have practices, we have tools but in the end, it's humans and how they act that do the difference.

**Review by matks — APPROVED** [review]

**Comment by sarahdib** [comment]
@sarjon  catalog -> categories grid the following elements are not working anymore
1- Select all
2- Drag and drop
3- Reset
4- refresh list
5- show sql query
6- export to sql manager

**Comment by sarjon** [comment]
Sorry @sarahdib, could you try rebuilding assets?

**Comment by sarahdib** [comment]
@sarjon thank you it's working :)

**Comment by matks** [comment]
Thank you @sarjon

**Pull request merged**